src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReport.java

```diff
@@ -0,0 +1,116 @@
+import uk.gov.hmcts.ethos.replacement.docmosis.helpers.ReportHelper;
+import uk.gov.hmcts.ethos.replacement.docmosis.reports.ReportParams;
+import java.util.Comparator;
+import java.util.List;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.CLAIMS_BY_HEARING_VENUE_REPORT;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.SINGLE_HEARING_DATE_TYPE;
+
+public class ClaimsByHearingVenueReport {
+    private final ClaimsByHearingVenueReportDataSource reportDataSource;
+    private final ReportParams reportParams;
+    private static final String NULL_STRING_VALUE = "Null";
+
+    public ClaimsByHearingVenueReport(ClaimsByHearingVenueReportDataSource dataSource, ReportParams params) {
+        reportDataSource = dataSource;
+        reportParams = params;
+    }
+
+    public ClaimsByHearingVenueReportData generateReport(String listingDate, String hearingDateType) {
+        var submitEvents = reportDataSource.getData(
+            UtilHelper.getListingCaseTypeId(reportParams.getCaseTypeId()),
+            reportParams.getDateFrom(), reportParams.getDateTo());
+        var claimsByHearingVenueReportData = initReport(reportParams.getCaseTypeId());
+        setReportListingDate(claimsByHearingVenueReportData, listingDate,
+            reportParams.getDateFrom(), reportParams.getDateTo(), hearingDateType);
+
+        if (CollectionUtils.isNotEmpty(submitEvents)) {
+            setReportData(submitEvents, claimsByHearingVenueReportData);
+        }
+        return claimsByHearingVenueReportData;
+    }
+
+    private ClaimsByHearingVenueReportData initReport(String caseTypeId) {
+        var reportData =  new ClaimsByHearingVenueReportData();
+        reportData.setOffice(UtilHelper.getListingCaseTypeId(caseTypeId));
+        reportData.setReportType(CLAIMS_BY_HEARING_VENUE_REPORT);
+        reportData.setDocumentName(CLAIMS_BY_HEARING_VENUE_REPORT);
+        return reportData;
+    }
+
+    private void setReportListingDate(ClaimsByHearingVenueReportData reportData, String listingDate,
+                                      String listingDateFrom, String listingDateTo, String hearingDateType) {
+        if (SINGLE_HEARING_DATE_TYPE.equals(hearingDateType)) {
+            reportData.setListingDate(listingDate);
+            reportData.setListingDateFrom(null);
+            reportData.setListingDateTo(null);
+            reportData.setHearingDateType(hearingDateType);
+            reportData.setReportPeriodDescription("On "
+                + UtilHelper.listingFormatLocalDate(ReportHelper.getFormattedLocalDate(listingDate)));
+        } else {
+            reportData.setListingDate(null);
+            reportData.setListingDateFrom(ReportHelper.getFormattedLocalDate(listingDateFrom));
+            reportData.setListingDateTo(ReportHelper.getFormattedLocalDate(listingDateTo));
+            reportData.setHearingDateType(hearingDateType);
+            reportData.setReportPeriodDescription("Between "
```

Comment:
assuming it is possible to run this report with a single date rather than a range, have you checked the description when there is a single date?

Reply:
Yes I have checked. It displays the description as intended/correctly when search is a single date type.